src/kpi.js

```diff
@@ -0,0 +1,47 @@
+import axios from 'axios';
+import { URL } from 'universal-url';
+
+export default function Kpi(kpiToken) {
+  const MAX_ATTEMPTS = 15 * 60; // 15 min
+
+  const delay = () => new Promise(res => setTimeout(res, 1000));
+
+  async function retryGet(url, attempt = 1) {
+    const res = await axios({
+      method: 'GET',
+      url,
+    });
+    const ifNoneMatch = new URL(url).searchParams.get('If-None-Match');
+    const ETag = res.headers.etag;
+    if (ETag === ifNoneMatch) {
+      if (attempt >= MAX_ATTEMPTS) {
+        return { data: [] };
+      }
+      await delay();
+      return retryGet(url, attempt + 1);
+    }
+    return res;
+  }
+
+  async function getResults(kpiId) {
+    try {
+      const { headers: { location: url } } = await axios({
+        headers: { Authorization: `TOKEN ${kpiToken}` },
+        method: 'GET',
+        url: `https://api.devicepilot.com/kpi/${kpiId}`,
+      });
+
+      const { data } = await retryGet(url);
+
+      return data;
+    } catch (err) {
+      console.error({ msg: 'error making request', err }); // eslint-disable-line no-console
+      throw new Error('KPI could not be fetched');
+    }
+  }
+
+  return {
+    retryGet,
```

Comment:
If you're not using it can you not export it please - it makes future refactors dififcult because the developer ends up checking to see if this (internal only) method is actually used anywhere.